### DISCUSSION TITLE
在confyui中使用的LoadImage，使用comfyui-client应该怎么上传图像呢

### DISCUSSION THREAD

**Discussion by ttidsn84320** [StableCanvas/comfyui-client#17] [Q&A] [answered]
在confyui中使用的LoadImage，使用comfyui-client应该怎么上传图像呢

**Comment by ttidsn84320** [comment]
const extensions = await this.client._enqueue_prompt(modifiedPrompt, {});
    我这里直接加载的导出的api工作流

**Comment by ttidsn84320** [comment]
confyui中有上传接口 看起来可以直接用 打扰了各位

**Comment by zhzLuke96** [comment, marked as answer]
这个问题确实 readme 里写得不是很清楚。我再解释一下。

你说的comfyui的上传接口，可能可以用，但是我不太推荐。
主要原因是，comfyui 本身对文件管理支持不完全，如果你要用它的文件接口可能导致你需要自己清理文件，不太优雅。
并且使用基于文件的方法，导致你需要使用 `LoadImage` 这个节点将读取本地文件，可能也会给你的服务引入一些代码风险。

所以，对于文件上传，或者更直接来说就是 `image to image` 和 `inpainting` 任务，我推荐的方式是使用 base64 来传递图片。

下面是一些指南：
- 为了支持 base64 文件传递，你需要安装一个自定义节点（在comfyui里）：
https://github.com/Acly/comfyui-tooling-nodes

- 如果你使用 pipeline 接口，那么可以直接使用 pipe.image() 传递数据（它内部就是依赖 comfyui-tooling-nodes ）
使用示例可以看这里
https://github.com/StableCanvas/comfyui-client/blob/main/examples/nodejs/src/main-pipe-i2i.ts
实现代码可以看这里
https://github.com/StableCanvas/comfyui-client/blob/main/src/pipeline/base.ts#L295-L297

- 如果你使用更底层的 workflow 接口，那么你可以参考上面的代码来构造工作流以传递你的 base64 图片
简单来说就是使用 `ETN_LoadImageBase64` 节点

- 如果你使用更更底层的直接 `client._enqueue_prompt` ，那么，你需要修改你的 workflow.json ，以使用 ETN_LoadImageBase64 节点，并且需要动态的修改这个 json object 把编码后的 base64 设置为节点的 image 输入值
> 不太推荐 `client._enqueue_prompt` ，你可以尝试用 [cli 工具](https://github.com/StableCanvas/comfyui-client/tree/main?tab=readme-ov-file#cli) 将 workflow 转换为 代码，然后使用代码来创建 workflow 而不是直接传递 api 数据

**Reply by ttidsn84320** [reply]
感谢回复，我修改为推荐的方式试试
